viam-cartographer/src/carto_facade_go/carto_facade.go

```diff
@@ -0,0 +1,278 @@
+func (vc *CViamCarto) Stop(ctx context.Context) error {
+	status := C.viam_carto_stop(vc.viamCarto)
+
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return errors.New(string(status))
+	}
+
+	return nil
+}
+
+// TerminateViamCarto calls viam_carto_terminate to clean up memory for viam carto
+func (vc *CViamCarto) TerminateViamCarto(ctx context.Context) error {
+	status := C.viam_carto_terminate(&vc.viamCarto)
+
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return errors.New(string(status))
+	}
+
+	return nil
+}
+
+// CViamCartoAddSensorReading is a struct to hold the c type viam_carto_sensor_reading
+type CViamCartoAddSensorReading struct {
+	vcasr C.viam_carto_sensor_reading
+}
+
+// AddSensorReading is a wrapper for viam_carto_add_sensor_reading
+func (vc *CViamCarto) AddSensorReading(ctx context.Context) error {
+	resp := CViamCartoAddSensorReading{vcasr: C.viam_carto_sensor_reading{}}
+
+	status := C.viam_carto_add_sensor_reading(vc.viamCarto, &resp.vcasr)
+
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return errors.New(string(status))
+	}
+
+	status = C.viam_carto_add_sensor_reading_destroy(&resp.vcasr)
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return errors.New(string(status))
+	}
+
+	return nil
+}
+
+// CViamCartoGetPositionResponse is a struct to hold the c type viam_carto_get_position_response
+type CViamCartoGetPositionResponse struct {
+	vcgpr C.viam_carto_get_position_response
+}
+
+// GetPosition is a wrapper for viam_carto_get_position
+func (vc *CViamCarto) GetPosition(ctx context.Context) (spatialmath.Pose, string, error) {
+	resp := CViamCartoGetPositionResponse{vcgpr: C.viam_carto_get_position_response{}}
+
+	status := C.viam_carto_get_position(vc.viamCarto, &resp.vcgpr)
+
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return nil, "", errors.New(string(status))
+	}
+
+	pos := r3.Vector{
+		X: float64(resp.vcgpr.x),
+		Y: float64(resp.vcgpr.y),
+		Z: float64(resp.vcgpr.z),
+	}
+
+	// TODO: What should be done with quat info?
+	// quat := &spatialmath.Quaternion{
+	// 	Real: float64(resp.vcgpr.real),
+	// 	Imag: float64(resp.vcgpr.imag),
+	// 	Jmag: float64(resp.vcgpr.jmag),
+	// 	Kmag: float64(resp.vcgpr.kmag)}
+
+	ori := &spatialmath.OrientationVectorDegrees{
+		OX: float64(resp.vcgpr.o_x),
+		OY: float64(resp.vcgpr.o_y),
+		OZ: float64(resp.vcgpr.o_z),
+	}
+
+	status = C.viam_carto_get_position_response_destroy(&resp.vcgpr)
+	if !successfulStatus(status) {
+		// TODO: how to make this error useful?
+		return nil, "", errors.New(string(status))
+	}
+
+	// TODO: fill component reference in?
+	return spatialmath.NewPose(pos, ori), "", nil
+}
+
+// CViamCartoGetPointCloudMapResponse is a struct to hold the c type viam_carto_get_point_cloud_map_response
+type CViamCartoGetPointCloudMapResponse struct {
+	vcgpcmr C.viam_carto_get_point_cloud_map_response
+}
```

Comment:
```suggestion
type CartoGetPointCloudMapResponse struct {
	value C.viam_carto_get_point_cloud_map_response
}
```

Lets also move this to the top of the file with the other structs